src/rate-limit-view/rate-limit-view.ts

```diff
@@ -0,0 +1,89 @@
+import COLORS from "../data/rate-limit-view-colors.json";
+
+interface RateLimitColors {
+	negative: string,
+	positive: string[]
+}
+
+const RATE_LIMIT_COLORS: RateLimitColors = COLORS;
+const RATE_LIMIT = 333;
+const COLOR_INTERVAL = 20;
+const COLOR_MAX_MS = RATE_LIMIT_COLORS.positive.length * COLOR_INTERVAL;
+
+export class RateLimitView {
+	static oldtime = Date.now();
+	static newtime = Date.now();
+	static bgColor = "";
+	static fontColor = "";
+	static isDelta = false;
+	static isEnabled = false;
+
+	constructor(Options: any) {
+		$("<span id=ratelim class=ratelimit>RL: <span id=rate></span></span>").insertBefore(".CANT_REPLY");
+		$(".ratelimit").css({
+			background: "transparent",
+			display: "none",
+			float: "right",
+			"font-size": "smaller",
+			"margin-left": "10px",
+			"margin-top": "5px",
+		});
+		this.initOptions(Options);
+	}
+
+	initOptions(Options: any): void {
+		Options.addSelect({
+			default: 0,
+			help: "Rate limit viewer by post. Not real, just estimate lol",
+			label: "RATE LIMIT VIEW",
+			onchange: RateLimitView.onChange.call(this),
+			options: ["Disabled", "Normal", "Delta"],
+			section: "Advanced 2",
+		});
+
+		$("#new-update-form .save-button button").on("click", RateLimitView.onMakeUpdate);
```

Comment:
This event handler should be the implementation detail of a custom update event so it can be reused elsewhere.

Reply:
This is on clicking the "make update" button, aren't update events currently only for received updates?

Reply:
In that case, it can be part of new event type with a different parameter instead of `UpdateInfo`. The other references should also be updated as part of this new event as well.

Reply:
That seems out of scope of this PR then and should be addressed separately.

I'd name the Element for now to make it a bit more readable.